### PR TITLE
Fix pagination issue and security settings refactoring

### LINKED ISSUE #378: Pagination Issue Through Webpack
For some reason, as soon as I add in the api url into webpack.common.js, the pagination goes away but when I am running in development it shows.
new webpack.DefinePlugin({
            'process.env': {
                NODE_ENV: `'${options.env}'`,
                BUILD_TIMESTAMP: `'${new Date().getTime()}'`,
                VERSION: `'${packageJson.version}'`,
                DEBUG_INFO_ENABLED: options.env === 'development',
                // --API Url
                SERVER_API_URL: `''`
            }
        }),
users

### PR DESCRIPTION
**Pull request by maznag**
Fix #378.

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/dotnetcore/src/Project/appsettings.json.ejs [hunk 1]
```diff
@@ -90,7 +90,8 @@
                       //}
                 }    
                 <%_ } _%>
-            }
+            },
+            "EnforceHttps":  false
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project/appsettings.json.ejs:94]: Mayve you can set EnforceHttps to false in https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/server/templates/dotnetcore/src/Project/appsettings.Development.json.ejs
and set to true in https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/server/templates/dotnetcore/src/Project/appsettings.Production.json.ejs

**Review by nicolas63 — CHANGES_REQUESTED** [review]

**Review by nicolas63 — APPROVED** [review]

**Comment by maznag** [comment]
@nicolas63 @ivanmonteiro I propose to refactor `appsettings` and simplify security settings section as follow (remove jhipster key):

```
"Security": {
        "Authentication": {
            "Jwt": {
                "Base64Secret": "bXktc2VjcmV0LWtleS13aGljaC1zaG91bGQtYmUtY2hhbmdlZC1pbi1wcm9kdWN0aW9uLWFuZC1iZS1iYXNlNjQtZW5jb2RlZAo=",
                "TokenValidityInSeconds": 86400,
                "TokenValidityInSecondsForRememberMe": 2592000
            }
        },
        "Cors": {
            "AllowedOrigins": "*",
            "AllowedMethods": "*",
            "AllowedHeaders": "*",
            "ExposedHeaders": "Authorization,Link,X-Total-Count,X-Pagination",
            "AllowCredentials": true,
            "MaxAge": 1800
        },
        "EnforceHttps": false
    }
```

and 

```
public class SecuritySettings
    {
        public Authentication Authentication { get; set; }
        public Cors Cors { get; set; }
        public bool EnforceHttps { get; set; }
    }
```

What do you think ?

**Comment by nicolas63** [comment]
@maznag yes i think it's good idea
The build error is due to format error can you verify ?

**Comment by ivanmonteiro** [comment]
> @nicolas63 @ivanmonteiro I propose to refactor `appsettings` and simplify security settings section as follow (remove jhipster key):
> 
> ```
> "Security": {
>         "Authentication": {
>             "Jwt": {
>                 "Base64Secret": "bXktc2VjcmV0LWtleS13aGljaC1zaG91bGQtYmUtY2hhbmdlZC1pbi1wcm9kdWN0aW9uLWFuZC1iZS1iYXNlNjQtZW5jb2RlZAo=",
>                 "TokenValidityInSeconds": 86400,
>                 "TokenValidityInSecondsForRememberMe": 2592000
>             }
>         },
>         "Cors": {
>             "AllowedOrigins": "*",
>             "AllowedMethods": "*",
>             "AllowedHeaders": "*",
>             "ExposedHeaders": "Authorization,Link,X-Total-Count,X-Pagination",
>             "AllowCredentials": true,
>             "MaxAge": 1800
>         },
>         "EnforceHttps": false
>     }
> ```
> 
> and
> 
> ```
> public class SecuritySettings
>     {
>         public Authentication Authentication { get; set; }
>         public Cors Cors { get; set; }
>         public bool EnforceHttps { get; set; }
>     }
> ```
> 
> What do you think ?

I agree with the change

**Pull request merged**